src/grasplan/place.py

```diff
@@ -358,12 +367,17 @@ def place_object(
                         rospy.loginfo(f'going to intermediate arm pose {arm_pose} to disentangle cable')
                         self.move_arm_to_posture(arm_pose)
 
-            rospy.loginfo(f'sending place {object_to_be_placed} goal to {self.place_object_server_name} action server')
-            action_client.send_goal(goal)
-            rospy.loginfo(f'waiting for result from {self.place_object_server_name} action server')
-            if action_client.wait_for_result(rospy.Duration.from_sec(self.timeout)):
+            rospy.loginfo(
+                f'sending place {object_to_be_placed} goal to {PLACE_OBJECT_SERVER_NAME} action server '
+                'and waiting for result'
+            )
+            if self.action_client_helper.send_goal_to_rogue_server_and_wait(goal, action_client, patience_timeout=0.1):
```

Comment:
This is a combination of "action_client.send_goal(goal)" and "action_client.wait_for_result" that checks at 10 hz if preemption was requested by user, this is done in a new file called action_client_helper.py . Additionally if MoveIt ignores the cancellation request, it forces it. patience_timeout is the amount of time to wait for MoveIt to acknowledge the cancellation. 0.1 is quite low but we are factoring in expert knowledge in the sense that we already know that our request will be ignored... and therefore it does not make sense to make this value higher.